sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -290,35 +350,28 @@ abstract Builder setWritableByteChannelFactory(
      *
      * <p>This default policy can be overridden using {@link #withFilenamePolicy(FilenamePolicy)},
      * in which case {@link #withShardNameTemplate(String)} and {@link #withSuffix(String)} should
-     * not be set.
+     * not be set. Custom filename policies do not automatically see this prefix - you should
+     * explicitly pass the prefix into your {@link FilenamePolicy} object if you need this.
+     *
+     * <p>If {@link #withTempDirectory} has not been called, this filename prefix will be used to
+     * infer a directory for temporary files.
```

Comment:
Agreed. How about doing the same for withFilenamePolicy? Just make it a to() override.